week_1/Array/move_zeroes.js

```diff
@@ -0,0 +1,22 @@
+// Given an integer array nums, move all 0's to the end of it while maintaining the relative order of the non-zero elements.
+
+// Note that you must do this in-place without making a copy of the array.
```

Comment:
This looks like a left over from the test instructions, we should rephrase this to be more of a warning than a footnote. Generally, at least in JavaScript, mutating params can be a _very_ bad thing and lead to _very_ strange behavior. It can be an effective memory savor when you are expecting it, though 😄 

For context, let's look at how Rust handles mutation. This function's signature would look something like this:
```rust
fn moveZeroes(&mut nums)
```

The `&mut` here tells the person who is about to use this function that mutation is more than likely going to happen (otherwise, it's just bad Rust code 😄 )

If you wanted to tell the caller that you _weren't_ going to mutate anything, the function signature would look like this:
```rust
fn moveZeros(&nums
```

The omission of `mut` means any code in the function can't mutate the param it was given! If you were to try, your code won't compile 🚀 

All of this makes it super clear when you plan on mutating and when you aren't. In languages like JavaScript where you can't express this that easily, you want to take the time to explicitly call out when you are mutating. Don't be afraid to be loud about it 😄 it's typically common, in a good code base, to assume things aren't mutating; with a some exceptions that come to mind. 

_note, none of that is real Rust code, trimmed off a couple things that weren't applicable 😄  but it's close!_